File: Kms.Security.Common/Domain/EnterprisePosition.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kms.Security.Common.Domain
{
   public class EnterprisePosition:IDomainBase<Guid>
    {
        protected EnterprisePosition()
        {
        }
        public EnterprisePosition(Guid id, String name, Guid? parentId, Guid organizationId,int? sortingNumber)
        {
            Id = id;
            Name = name;
            ParentId = parentId;
            OrganizationId = organizationId;
            this.SortingNumber = sortingNumber;
        }
        public void SetNewId()
        {
            this.Id = Guid.NewGuid();
        }

        public bool IdIsEmpty()
        {
            return Id == Guid.Empty;
        }

        public Guid Id { get; protected set; }
        public string Name { get; protected set; }
        public EnterprisePosition Parent { get; protected set; }
        public Guid? ParentId { get; protected set; }
        public Guid OrganizationId { get; protected set; }
        public Organization Organization { get; protected set; }

        public int? SortingNumber { get;protected set; }
    }
}

File: Kms.Security.Portal/app/nodak/common/enterprisePosition/enterprisePosition.ts
module nodak.common.models {

    export class EnterprisePosition extends nodak.models.ModelBase {

        constructor() {
            super(enums.SubSystems.Common, "EnterprisePosition");
        }
        AddBussinessModelValidation() {

        }
        private id: any
        set Id(value)
        { this.id = value; }
        get Id()
        { return this.id; }

        private name: string
        set Name(value)
        { this.name = value; }
        get Name()
        { return this.name; }

        private parent: EnterprisePosition
        set Parent(value: EnterprisePosition)
        { this.parent = value; }
        get Parent()
        { return this.parent; }

        private parentId: any
        set ParentId(value)
        { this.parentId = value; }
        get ParentId()
        { return this.parentId; }

        private organizationId: any
        set OrganizationId(value)
        { this.organizationId = value; }
        get OrganizationId()
        { return this.organizationId; }

        private organization:models.Organization
        set Organization(value)
        { this.organization = value; }
        get Organization()
        { return this.organization; }

        private sortingNumber;
        set SortingNumber(value) {
            this.sortingNumber = value;
        }
        get SortingNumber() {
            return this.sortingNumber;
        }

    }
}
